packages/form-js-viewer/src/render/components/FormField.js

```diff
@@ -98,14 +98,16 @@ export function FormField(props) {
 
   const onBlur = useCallback(() => {
 
+    const value = get(data, valuePath);
+
     if (initialValidationTrigger) {
       setInitialValidationTrigger(false);
       viewerCommands.updateFieldValidation(field, value, indexes);
     }
 
     eventBus.fire('formField.blur', { formField: field });
 
-  }, [ eventBus, field, indexes, value, viewerCommands, initialValidationTrigger ]);
+  }, [ eventBus, field, indexes, viewerCommands, initialValidationTrigger, data, valuePath ]);
```

Comment:
In the end the proper fix was just 2 lines long, I had a more complicated fix that was longer 🫠

Reply:
Yeah I considered doing this at some point but thought the value would repopulate through. Oh well, fair enough. This should ensure things don't get out of sync again.